### PR TITLE
Update Pagination Logic on Hubs

### PR DESCRIPTION
**Pull request by manveerxyz**
Had to make some changes to the backend to make pagination work properly.

- Ensure the `ordering` param is actually getting used
- Add "id" as an ordering field to break ties for hubs that have identical scores
- Added tests for ordering and pagination

### CODE REVIEW

File: src/hub/filters.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def filter(self, qs, value):
                 + F("paper_count")
             )
 
-            qs = qs.annotate(score=score).order_by("-score")
+            qs = qs.annotate(score=score).order_by(*value, "id")
```
Comment (lightninglu10) [inline src/hub/filters.py:43]: so we're ordering first by score and then by id?

Comment (manveerxyz) [reply src/hub/filters.py:43]: `id` is meant to be a tie-breaker if the scores are the same. This change is to resolve the issue we had with the same item showing up on multiple pages.

**Comment by manveerxyz** [comment]
Okay, closed https://github.com/ResearchHub/researchhub-backend/pull/1381 and removed sorting filters in this PR.

Also updated the caching in the view.

**Comment by lightninglu10** [comment]
Looks good, going to merge & deploy

**Pull request merged**